jenkins/PerfCI-Chaos/01-PerfCI-Chaos-OpenShift-Deployment/Jenkinsfile

```diff
@@ -0,0 +1,199 @@
+        QUAY_BENCHMARK_RUNNER_REPOSITORY = credentials('perfci_quay_repository')
+        WORKSPACE = '/home/jenkins'
+        PROVISION_PORT = 22
+        KUBEADMIN_PASSWORD_PATH = '/root/.kube/kubeadmin-password'
+        KUBECONFIG_PATH = '/root/.kube/config'
+        PROVISION_INSTALLER_PATH = '/root/jetlag/./run_jetlag.sh'
+        PROVISION_INSTALLER_CMD = 'pushd /root/jetlag;/root/jetlag/./run_jetlag.sh 1>/dev/null 2>&1;popd'
+        PROVISION_INSTALLER_LOG = 'tail -100 /root/jetlag/jetlag.log'
+        PRIVATE_KEY_PATH = '/home/jenkins/.ssh/provision_private_key'
+        CONFIG_PATH = '/home/jenkins/.ssh/config'
+        CONTAINER_PRIVATE_KEY_PATH = '/root/.ssh/provision_private_key'
+        CONTAINER_CONFIG_PATH = '/root/.ssh/config'
+        ANSIBLE_TMP_PATH = '/root/.ansible/tmp/'
+        PROVISION_TIMEOUT = 7200
+    }
+    stages {
+        stage('Cleanup') {
+            steps {
+                script {
+                    try {
+                        // Clean the Jenkins workspace
+                        echo "Cleaning Jenkins workspace"
+                        deleteDir()
+
+                        // Delete assisted installer previous deployments
+                        sh "curl http://${PROVISION_IP}:8090/api/assisted-install/v2/clusters | jq '.[].id' -r | xargs -I % curl -X DELETE http://${PROVISION_IP}:8090/api/assisted-install/v2/clusters/%"
+
+                        // remove ansible tmp files
+                        sh "sudo rm -rf ${ANSIBLE_TMP_PATH}"
+
+                        // Delete benchmark-runner images
+                        sh '''
+                            # Check if the image exists
+                            if [[ "$(sudo podman images -q ${QUAY_BENCHMARK_RUNNER_REPOSITORY} 2> /dev/null)" != "" ]]; then
+
+                                # Get containers using the image
+                                containers=$(sudo podman ps -a --filter ancestor=${QUAY_BENCHMARK_RUNNER_REPOSITORY} -q)
+
+                                if [[ "$containers" != "" ]]; then
+                                    echo "Stopping and removing containers using the image..."
+                                    sudo podman stop $containers
+                                    sudo podman rm -f $containers
+                                fi
+
+                                # Now remove the image
+                                sudo podman rmi -f $(sudo podman images -q ${QUAY_BENCHMARK_RUNNER_REPOSITORY} 2> /dev/null)
+                            fi
+                        '''
+                        try {
+                              // Remove assisted-installer containers - for apply new version in jetlag
+                              sh '''
+                                    sudo podman ps | awk '{print $1}' | xargs -I % sudo podman stop % || true;
+                                    sudo podman ps -a | awk '{print $1}' | xargs -I % sudo podman rm % || true;
```

Comment:
Use `podman rmi`

Reply:
Removes the containers using their IDs and not the images

Reply:
My bad, I meant `podman  rm -f` rather than just `podman rm`.